File: src/main/java/com/example/hnefatafl/context/RequestContext.java
package com.example.hnefatafl.context;

public class RequestContext {
    private static final ThreadLocal<String> requestIdHolder = new ThreadLocal<>();

    public static void setClientId(String clientId) {
        requestIdHolder.set(clientId);
    }

    public static String getClientId() {
        return requestIdHolder.get();
    }

    public static void clear() {
        requestIdHolder.remove();
    }
}

File: src/main/java/com/example/hnefatafl/context/RequestContext.kt
package com.example.hnefatafl.context

object RequestContext {
    private val requestIdHolder = ThreadLocal<String>()
    private val requestCorrelationIdHolder = ThreadLocal<String>()
    @JvmStatic
    var clientId: String
        get() = requestIdHolder.get()
        set(clientId) {
            requestIdHolder.set(clientId)
        }

    @JvmStatic
    var correlationId: String
        get() = requestCorrelationIdHolder.get()
        set(correlationId) {
            requestCorrelationIdHolder.set(correlationId)
        }

    @JvmStatic
    fun clear() {
        requestIdHolder.remove()
        requestCorrelationIdHolder.remove()
    }
}